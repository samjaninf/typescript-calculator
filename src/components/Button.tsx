import { BoundAction } from '../types'
import { StyleAttribute, css } from 'glamor'
import { button } from '../styles'
import { h } from 'preact'

type ButtonProps = {
  action: BoundAction,
  className?: StyleAttribute,
  text: string,
}

const handleClick = ( action: BoundAction ) => (): void => {
  action()
}

const Button = ({ action, className, text }: ButtonProps): JSX.Element => (
  <button
    className={css(button, className).toString()}
    dangerouslySetInnerHTML={{ __html: text }}
    onClick={handleClick(action)}
  />
)

export default Button
