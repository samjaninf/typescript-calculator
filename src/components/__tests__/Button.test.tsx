import Button from '../Button'
import snapshot from './utils/snapshot'

describe('Button', () => {
  it('renders', () => snapshot(Button))
  it.skip('sets onClick property', () => snapshot(Button, { action: () => ({ type: 'ADD' }) }))
  it('sets className property', () => snapshot(Button, { className: 'css-3lj4th1' }))
  it('sets text property', () => snapshot(Button, { text: '.' }))
  it('sets text property with HTML entities', () => snapshot(Button, { text: '&minus;' }))
})
