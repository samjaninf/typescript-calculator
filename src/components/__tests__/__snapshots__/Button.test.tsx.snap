// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `"<button class=\\"css-1ht4jl3\\"></button>"`;

exports[`Button sets className property 1`] = `"<button class=\\"css-18ir7kq\\"></button>"`;

exports[`Button sets text property 1`] = `"<button class=\\"css-1ht4jl3\\">.</button>"`;

exports[`Button sets text property with HTML entities 1`] = `"<button class=\\"css-1ht4jl3\\">&minus;</button>"`;
